Feature/frontend/chat list

juykoung opened the pull request:
1. ChatRoom과 ChatListScreen의 공통 로직 분리 및 위임
채팅방 나가기, 삭제, 시간 포맷팅 등 비동기 상태 변화가 적거나 순수한 로직은 외부 util로 분리 후 각 컴포넌트에서 호출하는 방식으로 변경

2. ChatRoom의 파일 로직 분리
- 목적 : 컴포넌트 역할을 UI와 상태 관리로 축소 및 FileBox에서 재사용할 수 있도록 로직 계층 분리
- **계층 분리**
-API 계층 : 파일 업로드/조회 서버 통신
-유틸리티 계층 : 순수 다운로드 실행, 날짜별 그룹화

juykoung committed:
refactor: ChatList 관련 폴더구조 변경

juykoung committed:
refactor: ChatList와 ChatRoom에 겹치는 기능 분리

- FriendSelectionModal 컴포넌트 생성 및 적용 (친구 선택 로직 통합)
- Sidebar 컴포넌트 분리 (메인 탐색 UI 재사용)

SeungminShin97 committed:
Merge pull request #19 from side-pingpong/feature/frontend/base

Feature/frontend/base

juykoung committed:
머지 충돌 해결

juykoung committed:
중복파일 삭제

juykoung committed:
chore: 파일 경로 수정

juykoung committed:
feat: NewChatModal 상태 끌어올리기 적용

- FriendSelectionModal에 onSelectionChange prop을 추가하여 선택 상태를 NewChatModal로 끌어올림.
- NewChatModal에서 선택된 친구 수(selectedFriendIds.length)를 기반으로 채팅방 이름 입력창을 조건부 렌더링하도록 수정

juykoung committed:
remove: NewChatModal 레파지토리 히스토리에서 삭제

juykoung committed:
refactor: 채팅 관련 공통 로직 유틸리티로 분리 및 재사용성 확보

TimeUtils.js:
- formatDate, isSameDay 등 날짜 포맷 함수를 분리.
- formatChatTime 구현해 ChatListItem의 시간 표시 UX를 개선

ChatUtils.js:
- handleChatRoomLeave, handleDeleteChatRoom 등 채팅방 관리 액션을 분리해 각 페이지에서 사용

juykoung committed:
chore: 파일 이름 변경

juykoung committed:
refactor: 사이드바 라우팅 도입

Sidebar를 상태 전환 방식에서 `react-router-dom`의 Maps를 사용한 경로 기반 페이지 이동 방식으로 변경
채팅 목록과 파일함 간의 페이지 전환 기능 구현

juykoung committed:
refactor: 채팅/파일 관련 로직 계층 분리

- fileApi 서버 통신(파일 업로드/조회) 담당 로직 분리
- fileUtils 순수 클라이언트 로직(다운로드 실행, 날짜별 그룹화) 분리
- ChatRoom에서 API 호출 및 상태 관리 담당

juykoung committed:
chore: 미사용 import, 컴포넌트 제거

judy78799 commented on frontend/src/pages/ChatRoom.js:
```diff
@@ -735,43 +725,14 @@ export default function ChatRoom() {
+                    onConfirm={handleInviteFriends} // 선택된 ID를 처리하는 함수 전달
+                    title="멤버 초대하기"
+                    confirmLabel="초대하기"
+                />
```
FriendSelectionModal -> UI + 로직 분리 리팩토링 확인했습니다.

judy78799 commented on frontend/src/pages/chatList/ChatListItem.js:
```diff
@@ -13,7 +13,7 @@ const ChatListItem = React.memo(({ chat, onContextMenu }) => {
             <div className="mr-4">
                 <div className="w-12 h-12 bg-gray-600 rounded-full flex items-center justify-center text-xl">
-                    {profileEmoji}
+                    {profileImage}
```
이모지 -> 이미지 확인이용

bill291104 committed:
chore: 보안 관련된 정보들은 application-secret.yml 에서 관리

- application-secret.yml 에서 api key 같은 민감 정보들을 통합 관리합니다.
- application-secret.yml 을 .gitignore 에 추가합니다.
- application-secret.tml 과 google credential json 파일은 따로 공유합니다.

juykoung committed:
chore: 민감 정보 설정 파일 Git 추적 제거

juykoung committed:
chore: dev 최신 병합 및 충돌 해결

juykoung committed:
feat: 파일업로드 진행률 반영
